File: Api/AvaluxAuth.DataAccess/Entities/ApplicationEntity.cs
using System.ComponentModel.DataAnnotations;

namespace AvaluxAuth.DataAccess.Entities;

internal class ApplicationEntity
{
    public required Guid Id { get; init; }

    [MaxLength(30)] public required string Name { get; init; }
    [MaxLength(32)] public required string ClientId { get; init; }
    [MaxLength(32)] public required string ClientSecret { get; init; }
    public string[] RedirectUrls { get; init; } = [];

    public required DateTime CreatedAt { get; init; }
    public DateTime? DeletedAt { get; init; }

    public ICollection<ProviderEntity> Providers { get; init; } = [];
}

File: UI/src/app/entities/application-entity.ts
import {Moment} from 'moment';

export interface ApplicationEntity {
  id: string;
  clientId: string;
  clientSecret: string;
  parameters: ApplicationParametersEntity;
  createdAt: Moment;
  deletedAt: Moment | null;
}

export interface ApplicationParametersEntity {
  name: string;
  redirectUrls: string[];
}
